python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -220,29 +220,31 @@ def unpack(
         return None
 
 
-@whitelist_for_serdes(
-    field_serializers={"start": DatetimeFieldSerializer, "end": DatetimeFieldSerializer}
-)
 class TimeWindow(NamedTuple):
     """An interval that is closed at the start and open at the end.
 
     Attributes:
-        start (datetime): A pendulum datetime that marks the start of the window.
-        end (datetime): A pendulum datetime that marks the end of the window.
+        start (datetime): A datetime that marks the start of the window.
+        end (datetime): A datetime that marks the end of the window.
     """
 
     start: PublicAttr[datetime]
     end: PublicAttr[datetime]
 
-    @property
-    def is_empty(self) -> bool:
-        return self.start == self.end
 
-    @staticmethod
-    def empty() -> "TimeWindow":
-        return TimeWindow(start=datetime.max, end=datetime.max)
+@whitelist_for_serdes(
+    storage_name="TimeWindow",  # For back-compat with existing serdes
+    field_serializers={"start": DatetimeFieldSerializer, "end": DatetimeFieldSerializer},
```

Comment:
An example of something that I would like to do here once this is separated is to expose the underlying TimestampWithTimezone object more directly on this object - so instead of going from timestamp => datetime => back to timestamp, we can just start with the underlying timestamp. The fact that we are required to expose "start" as a public property *and* expose TimeWindow as a NamedTuple as part of the public API currently makes this impossible